src/java.base/share/classes/java/util/Arrays.java

```diff
@@ -3582,6 +3590,9 @@ public static short[] copyOf(short[] original, int newLength) {
      * @since 1.6
      */
     public static int[] copyOf(int[] original, int newLength) {
+        if (newLength == original.length) {
+            return original.clone();
```

Comment:
I am curious about the use of `clone ` for some primitive array types and `copyOf` using `System.arraycopy` in other types (e.g. `byte[]`). Do these types optimize differently or hit different intrinsics depending on primitive type? Is there difference in array zeroing?

Reply:
I went back and forth on this and also saw a small win using `arraycopy`, but the PR ended up in an inconsistent state with some using one and some using the other. While this discrepancy seem like something we should treat as a bug, I've arranged to use `copyOf` helper consistently for now.